lib/internal/net.js

```diff
@@ -5,7 +5,8 @@ module.exports = { isLegalPort };
 // Check that the port number is not NaN when coerced to a number,
 // is an integer and that it falls within the legal range of port numbers.
 function isLegalPort(port) {
-  if (typeof port === 'string' && port.trim() === '')
+  if ((typeof port !== 'number' && typeof port !== 'string') ||
+      (typeof port === 'string' && port.trim().length === 0))
     return false;
-  return +port === (port >>> 0) && port >= 0 && port <= 0xFFFF;
+  return +port === (+port >>> 0) && port <= 0xFFFF;
```

Comment:
Is `+` in `>>>` expression necessary? Internally the it will be converted to number, right?